### ISSUE TITLE
NetworkReachabilityManager issues with stopListening() and startListening() combination

### ISSUE THREAD

**Issue by zivkovic** [Alamofire/Alamofire#2731]
The following code works differently in Alamofire 4.7.0 and 4.8.1:
```
for manager in reachabilityManagers {
    manager.stopListening()
    manager.startListening()
 }
```
reachabilityManagers is a list containing all created managers(NetworkReachabilityManager).

## What did you do?
pod update

## What did you expect to happen?

Alamofire 4.7.0 and up to including 4.8.0 works correctly. When the above code is called, the manager stops listening and starts listening again. Right after it starts listening, it detects that network connection is present and all is good

## What happened instead?

Alamofire 4.8.1 works incorrectly. When the same above code is called, the manager stops listening, the function to start listening is called and returns 'true'(should work), but the code never(15 sec) realizes the phone has network connection meaning the listener never gets called.

The bug seems to be introduced in Alamofire 4.8.1

## Alamofire Environment

**Alamofire version:** 4.8.1
**Xcode version:** 10.1
**Swift version:** 4.2.1
**Platform(s) running Alamofire:** iPhone 7+
**macOS version running Xcode:** macOS 10.14.1

## Demo Project
I have no demo project for now. If time will allow, I will try to make it tomorrow or next week if required.

If you require any more info, let me know.

**Comment by jshier** [comment]
@cnoon Can you take a look at this when you get a chance?

**Comment by jshier** [comment]
In my refactor of reachability for Alamofire 5, I've found the cause of this issue: the change in 4.8.1 removed a state reset that was allowing the `startListening` call to fire the listener again. We'll see if we can get a fix in as part of 4.9.

**Comment by jshier** [comment]
This is fixed by #2931, which will be part of 4.9.0.
